holiday_defs_au.go

```diff
@@ -0,0 +1,129 @@
+// (c) 2014 Rick Arnold. Licensed under the BSD license (see LICENSE).
+
+package cal
+
+import (
+	"time"
+)
+
+// Holidays in Australia
+var (
+	AUNewYear                    = NewHolidayFunc(calculateNewYear)
+	AUAustralianDay              = NewHoliday(time.January, 26)
+	AUGoodFriday                 = GoodFriday
+	AUChristmasDay               = NewHolidayFunc(calculateChristmasDay)
+	AUBoxingDays                 = Christmas2
+	AUObservedChristmasBoxingDay = NewHolidayFunc(calculateChristmasDayBoxingDay)
+	AUEasterMonday               = EasterMonday
+	AUAnzacDay                   = NewHolidayFunc(calculateAnzacDay)
+)
+
+// AddAustralianHolidays adds all Australian holidays
+func AddAustralianHolidays(c *Calendar) {
+	c.AddHoliday(
+		AUNewYear,
+		AUAustralianDay,
+		AUGoodFriday,
+		AUEasterMonday,
+		AUAnzacDay,
+		AUChristmasDay,
+		AUBoxingDays,
+		AUObservedChristmasBoxingDay,
+	)
+}
+
+//  Holidays associated with the start of the modern Gregorian calendar.
+//
+//  New Year's Day is on January 1 and is the first day of a new year in the Gregorian calendar,
+//  which is used in Australia and many other countries. Due to its geographical position close
+//  to the International Date Line, Australia is one of the first countries in the world to welcome the New Year.
+//  If it falls on a weekend an additional public holiday is held on the next available weekday.
+//
+//  https://www.timeanddate.com/holidays/australia/new-year-day
+func calculateNewYear(year int, loc *time.Location) (time.Month, int) {
```

Comment:
These functions will end up being global for holidays in the cal package. I think it would be better to create a "nextWeekday()" function or something like that which would be generically useful.

Reply:
I updated the name of the function in calculateNewYearOceania because is actually a logic that is applied in NZ and Australia. Then the functionality to check for the next day is extracted in a new method. The rule in both countries are the same. Maybe we can find a better name that are calculating new year holidays in this way